### PR TITLE
Socint 221 cc UI cucumber tests

### PR DESCRIPTION
**Pull request by SteveScorfield**
This is for ticket https://collaborate2.ons.gov.uk/jira/browse/SOCINT-221

### CODE REVIEW

**Review by rpcatling — CHANGES_REQUESTED** [review]
some comments

File: src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java [hunk 1]
```diff
@@ -32,17 +44,41 @@ public void deleteDriver() {
     super.destroyPubSub();
   }
 
-  @Given("A case exists with my address")
-  public void createKnownCase() throws Exception {
+  private void sendInboundEvents() throws Exception {
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java:47]: I moved it to the bottom.

File: src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java [hunk 2]
```diff
@@ -14,13 +15,24 @@
...
 import uk.gov.ons.ctp.integration.cccucumber.selenium.pageobject.AvailableCases;
+import uk.gov.ons.ctp.integration.cccucumber.selenium.pageobject.NoCasesFound;
 import uk.gov.ons.ctp.integration.cccucumber.selenium.pageobject.SelAddressSelection;
 import uk.gov.ons.ctp.integration.cccucumber.selenium.pageobject.SelPostcodeSearch;
 import uk.gov.ons.ctp.integration.cccucumber.selenium.pageobject.SurveyEnquiryLine;
 
 public class CcSteps extends StepsBase {
 
+  private String currentPage;
+  private String caseNotFoundErrorMsg =
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java:28]: Agreed. These have been moved over.

File: src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java [hunk 3]
```diff
@@ -14,13 +15,24 @@
...
+      "Confirm this is the correct address, and that the it is the address "
+          + "the caller was asked to complete the survey for";
+
+  private String addressNotFound =
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/java/uk/gov/ons/ctp/integration/cccucumber/glue/CcSteps.java:32]: Agreed. These have been moved over.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 4]
```diff
@@ -1,10 +1,38 @@
 Feature:  Survey Enquiry Line
 
   @SelPostcodeSearch @Setup @TearDown
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:3]: Forgot to add the tags. They are now added.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
+    And The user selects the "Find Case via postcode" option
+    When The user is presented with callers postcode search option page
+    Then The user enters the callers postcode
+    And The user clicks "Continue"
+    Then The system displays a list of addresses for the postcode
+    And The user selects the callers address from the list of addresses which is number "6"
+    Then The user clicks "Continue"
+    Then The system displays the case details for the callers address
 
+  @SelPostcodeSearch @Setup @TearDown
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:15]: As above.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 6]
```diff
@@ -1,10 +1,38 @@
...
+    When The user is presented with callers postcode search option page
+    Then The user enters the callers postcode
+    And The user clicks "Continue"
+    Then The system displays a list of addresses for the postcode
+    And The user selects the callers address from the list of addresses which is number "6A"
+    Then The user clicks "Continue"
+    Then CCSvc returns no case for the selected address
+    And The user is presented with a no cases message
+
+  @SelPostcodeSearch @Setup @TearDown
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:28]: As above.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 7]
```diff
@@ -1,10 +1,38 @@
...
 
   @SelPostcodeSearch @Setup @TearDown
-  Scenario: SEL Postcode search
-  	Given A case exists with my address
-    And I have navigated to the SEL Postcode search page
-    When I have entered a UK postcode and clicked Continue
-    And I select the desired address from a list
-    Then The case is displayed for selection
+  Scenario: SEL Postcode search - A case exists with users address
+  	Given The SEL operator is in CCUI find case page
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:5]: First 6 are now as part of initial setup.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 8]
```diff
@@ -1,10 +1,38 @@
...
+    Then The user enters the callers postcode
+    And The user clicks "Continue"
+    Then The system displays a list of addresses for the postcode
+    And The user selects the callers address from the list of addresses which is number "6"
+    Then The user clicks "Continue"
+    Then The system displays the case details for the callers address
 
+  @SelPostcodeSearch @Setup @TearDown
+  Scenario: SEL Postcode search - A case does not exist with users address
+    Given The SEL operator is in CCUI find case page
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:17]: First 6 are now as part of initial setup.

File: src/test/resources/features/SurveyEnquiryLine.feature [hunk 9]
```diff
@@ -1,10 +1,38 @@
...
+    And The user clicks "Continue"
+    Then The system displays a list of addresses for the postcode
+    And The user selects the callers address from the list of addresses which is number "6A"
+    Then The user clicks "Continue"
+    Then CCSvc returns no case for the selected address
+    And The user is presented with a no cases message
+
+  @SelPostcodeSearch @Setup @TearDown
+  Scenario: SEL Postcode search - Cannot find address for postcode
+    Given The SEL operator is in CCUI find case page
```
[the remark this answers is not in the document]
Comment (SteveScorfield) [reply src/test/resources/features/SurveyEnquiryLine.feature:30]: First 6 are now as part of initial setup.

**Comment by rpcatling** [comment]
A few more things to change as discussed on the slack channel
- make hook for setup rather than private method , and update feature file accordingly
- fix naming convention for static final Strings
- add unique default CaseID UUID to avoid conflict with RHcuc tests

**Review by rpcatling — APPROVED** [review]

**Review by PeterBochel — APPROVED** [review]

**Pull request merged**